gratipay/utils/team_history.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from datetime import datetime
+from decimal import Decimal as D
+
+
+def get_end_of_year_totals(db, team, year):
+
+    received = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Received
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-team';
+    """, locals() )
+
+    if received is None:
+       received = D(0.00)
+
+    distributed = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Distributed
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-participant';
+    """, locals())
+
+    if distributed is None:
+        distributed = D(0.00)
+
+    return received, distributed
+
+
+def iter_team_payday_events(db, team, year=None):
+    """Yields payday events for the given Team.
+    """
+    current_year = datetime.utcnow().year
+    year = year or current_year
+
+    payments = db.all("""
+        SELECT payments.*, paydays.ts_start
+          FROM payments, paydays
+         WHERE payments.payday = paydays.id
+           AND team=%(team)s
+           AND extract(year from timestamp) = %(year)s
+         ORDER BY payments.payday DESC, direction, amount DESC
+       """, dict(team=team.slug, year=year), back_as=dict)
+
+    events = []
+    payday_id = payments[0]['payday']
+    payday_date = payments[0]['ts_start']
+    payday_events = []
+
+    if not payments:
+        return payments
+
+    #return payments
+
+    for payment in payments:
+        if payment['payday'] != payday_id:
```

Comment:
The following code (this is just an attempt at understanding the structure of the resulting object) should in theory produce a list with the `payday_id`, `payday_date` and list out all the transactions for that payday (unformatted) 
`{% for payday in events %}
    {% set id = payday['id'] %}
    {% set paydate = payday['date'] %}
    {% set pay_events = payday['events'] %}
    {% print id %}
    <br/>
    {% print paydate %}
    <br/>
    {% for each in pay_events %}
        {% print each %}
    {% endfor %}
    <br/>
    <br/>

{% endfor %}`

But it does not it produces the following 
![screen shot 2017-03-22 at 7 50 09 pm](https://cloud.githubusercontent.com/assets/1266505/24225877/5cb386a6-0f39-11e7-868a-bb45ae319b75.png)

Any suggestions???
Am I doing something wrong that I can't see.